Source/WebCore/rendering/mathml/RenderMathMLFraction.cpp

```diff
@@ -42,7 +42,7 @@ namespace WebCore {
 WTF_MAKE_ISO_ALLOCATED_IMPL(RenderMathMLFraction);
 
 RenderMathMLFraction::RenderMathMLFraction(MathMLFractionElement& element, RenderStyle&& style)
-    : RenderMathMLBlock(Type::MathMLFraction, element, WTFMove(style))
+    : RenderMathMLRow(Type::MathMLFraction, element, WTFMove(style))
```

Comment:
RenderMathMLRow overrides the following methods:


```
    void layoutBlock(bool relayoutChildren, LayoutUnit pageLogicalHeight = 0_lu) override;
    std::optional<LayoutUnit> firstLineBaseline() const override;
    void computePreferredLogicalWidths() override;
    ASCIILiteral renderName() const override { return "RenderMathMLRow"_s; }
 ```

* If I'm correct, there are overridden methods in RenderMathMLFraction/RenderMathMLRoot/RenderMathMLScripts, except RenderMathMLRoot::firstLineBaseline(), please add one.

* I believe renderName() never calls the method of the parent class, so nothing to do.

* For layoutBlock(), firstLineBaseline() and computePreferredLogicalWidths() you should make sure that we call RenderMathMLRow's method if the renderer is not valid and run the class's implementation otherwise.

Reply:
Added RenderMathMLRoot::firstLineBaseline(). 3rd bullet point *should* be enforced